File: app/src/main/java/com/danshrout/thefoodapp/database/Recipe.java
package com.danshrout.thefoodapp.database;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "recipe_table")
public class Recipe {

    @PrimaryKey(autoGenerate = true)
    private int id;

    private String title;
    private String ingredients;
    private String directions;

    //Constructor
    public Recipe(String title, String ingredients, String directions) {
        this.title = title;
        this.ingredients = ingredients;
        this.directions = directions;
    }

    //Setter
    public void setId(int id) {
        this.id = id;
    }

    //Getter
    public int getId() {
        return id;
    }
    public String getTitle() {
        return title;
    }
    public String getIngredients() {
        return ingredients;
    }
    public String getDirections() {
        return  directions;
    }


}

File: app/src/main/java/com/danshrout/thefoodapp/models/Recipe.kt
package com.danshrout.thefoodapp.models

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "recipe_table")
data class Recipe(

    @PrimaryKey(autoGenerate = true)
    var recipeId: Int,

    @ColumnInfo(name = "image")
    val image: Int,

    @ColumnInfo(name = "title")
    val title: String,

    @ColumnInfo(name = "description")
    val description: String
)
